packages/block-editor/src/components/block-list/index.js

```diff
@@ -53,7 +49,7 @@ export default function BlockList( { className, __experimentalLayout } ) {
 	}, [] );
 
 	return (
-		<BlockNodes.Provider value={ blockNodes }>
+		<>
 			{ insertionPoint }
 			<BlockPopover />
```

Comment:
After this and #31134, we can move the popover and insertion point out of the BlockList component.